src/styles/getTooltipStyles.js

```diff
@@ -0,0 +1,156 @@
+    const tooltipRightEdge = componentCenter + (tooltipWidth / 2);
+
+    if (tooltipLeftEdge < GUTTER_WIDTH) {
+        // Tooltip is in left gutter, shift right by a multiple of four.
+        return roundToNearestMultipleOfFour(GUTTER_WIDTH - tooltipLeftEdge);
+    }
+
+    if (tooltipRightEdge > (windowWidth - GUTTER_WIDTH)) {
+        // Tooltip is in right gutter, shift left by a multiple of four.
+        return roundToNearestMultipleOfFour(windowWidth - GUTTER_WIDTH - tooltipRightEdge);
+    }
+
+    // Tooltip is not in the gutter, so no need to shift it horizontally
+    return 0;
+}
+
+/**
+ * Generate styles for the tooltip component.
+ *
+ * @param {Number} interpolatedSize - The current size of the tooltip used in the scaling animation.
+ * @param {Number} windowWidth - The width of the window.
+ * @param {Number} xOffset - The distance between the left edge of the window
+ *                           and the left edge of the wrapped component.
+ * @param {Number} yOffset - The distance between the top edge of the window
+ *                           and the top edge of the wrapped component.
+ * @param {Number} componentWidth - The width of the wrapped component.
+ * @param {Number} componentHeight - The height of the wrapped component.
+ * @param {Number} tooltipWidth - The width of the tooltip itself.
+ * @param {Number} tooltipHeight - The height of the tooltip itself.
+ * @returns {Object}
+ */
+export default function getTooltipStyles(
+    interpolatedSize,
+    windowWidth,
+    xOffset,
+    yOffset,
+    componentWidth,
+    componentHeight,
+    tooltipWidth,
+    tooltipHeight,
+) {
+    // Determine if the tooltip should display below the wrapped component.
+    // If a tooltip will try to render within GUTTER_WIDTH logical pixels of the top of the screen,
+    // we'll display it beneath its wrapped component rather than above it as usual.
+    const shouldShowBelow = (yOffset - tooltipHeight) < GUTTER_WIDTH;
+
+    // Determine if we need to shift the tooltip horizontally to prevent it
+    // from displaying too near to the edge of the screen.
+    const horizontalShift = computeHorizontalShift(windowWidth, xOffset, componentWidth, tooltipWidth);
+
+    return {
+        animationStyle: {
+            transform: [{
+                scale: interpolatedSize,
+            }],
+        },
+        tooltipWrapperStyle: {
+            position: 'absolute',
+            backgroundColor: themeColors.heading,
+            borderRadius: variables.componentBorderRadiusSmall,
+            ...spacing.pv1,
+            ...spacing.ph2,
+
+            // Shift the tooltip up by...
+            //   11 (font height)
+            // + 4 (top padding)
+            // + 4 (bottom padding)
+            // + 4 (upward shift of the pointer)
+            // + 3 (pointer height - 1)
+            // = 26
+            // OR
+            // Shift the tooltip down by...
+            // (component height) + (pointer height - 1)
```

Comment:
Maybe this would be clearer if this was also a variable with an explanation about how we came up with this number? If these values were to change would we need to recalculate the `26` again?

Reply:
Same goes for the `3`. Maybe would be better as a constant.